From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 31 Dec 2021 02:03:55 +0500
Subject: [PATCH] Global eula file


diff --git a/src/main/java/net/minecraft/server/Eula.java b/src/main/java/net/minecraft/server/Eula.java
index df44790b1c342d0a6ff794bea86319fa7498cbae..d7e6743cf981086f3ae88b200ae024431d997388 100644
--- a/src/main/java/net/minecraft/server/Eula.java
+++ b/src/main/java/net/minecraft/server/Eula.java
@@ -16,12 +16,24 @@ public class Eula {
 
     public Eula(Path eulaFile) {
         this.file = eulaFile;
-        this.agreed = SharedConstants.IS_RUNNING_IN_IDE || this.readFile();
+        this.agreed = SharedConstants.IS_RUNNING_IN_IDE || globalEula() || this.readFile(this.file);
     }
 
-    private boolean readFile() {
+    // Sapphire start - Global eula file
+    private boolean globalEula() {
+        java.io.File globalEula = new java.io.File(System.getProperty("user.home"), "eula.txt");
+        if (globalEula.exists()) {
+            return this.readFile(globalEula.toPath());
+        } else {
+            LOGGER.info("No global eula found at " + globalEula.getAbsolutePath());
+        }
+        return false;
+    }
+    // Sapphire end
+
+    private boolean readFile(java.nio.file.Path path) {
         try {
-            InputStream inputStream = Files.newInputStream(this.file);
+            InputStream inputStream = Files.newInputStream(path);
 
             boolean var3;
             try {
