From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 3 Dec 2021 21:18:54 +0500
Subject: [PATCH] Add GameProfileLookupEvent


diff --git a/src/main/java/com/destroystokyo/paper/profile/CraftPlayerProfile.java b/src/main/java/com/destroystokyo/paper/profile/CraftPlayerProfile.java
index 3ff790cec1ad89caec4be64421dd7d51652be598..620a273d0a5d5fd5e8afe7c95e4460b9f6620ba0 100644
--- a/src/main/java/com/destroystokyo/paper/profile/CraftPlayerProfile.java
+++ b/src/main/java/com/destroystokyo/paper/profile/CraftPlayerProfile.java
@@ -226,12 +226,23 @@ public class CraftPlayerProfile implements PlayerProfile, SharedPlayerProfile {
         MinecraftServer server = MinecraftServer.getServer();
         boolean isCompleteFromCache = this.completeFromCache(true, onlineMode);
         if (onlineMode && (!isCompleteFromCache || textures && !hasTextures())) {
-            GameProfile result = server.getSessionService().fillProfileProperties(profile, true);
-            if (result != null) {
-                copyProfileProperties(result, this.profile, true);
-            }
-            if (this.profile.isComplete()) {
-                server.getProfileCache().add(this.profile);
+            // Sapphire start - call GameProfileLookupEvent
+            io.sapphiremc.sapphire.api.event.GameProfileLookupEvent event = new io.sapphiremc.sapphire.api.event.GameProfileLookupEvent(
+                !org.bukkit.Bukkit.isPrimaryThread(), profile.getId(), profile.getName()
+            );
+            org.bukkit.Bukkit.getPluginManager().callEvent(event);
+            GameProfile eventProfile = event.getGameProfile();
+            if (eventProfile != null) {
+                this.profile = eventProfile;
+            } else {
+                GameProfile result = server.getSessionService().fillProfileProperties(profile, true);
+                if (result != null) {
+                    copyProfileProperties(result, this.profile, true);
+                }
+                if (this.profile.isComplete()) {
+                    server.getProfileCache().add(this.profile);
+                }
+                // Sapphire end
             }
         }
         return profile.isComplete() && (!onlineMode || !textures || hasTextures());
diff --git a/src/main/java/net/minecraft/world/level/block/entity/SkullBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/SkullBlockEntity.java
index e017020b7630f54c506d93ab0504c6f3299fc80b..07ebdf82ee9dc460391b1b4a2b50f105fd28d716 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/SkullBlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/SkullBlockEntity.java
@@ -139,7 +139,18 @@ public class SkullBlockEntity extends BlockEntity {
                     Util.ifElse(profile, (profilex) -> {
                         Property property = Iterables.getFirst(profilex.getProperties().get("textures"), (Property)null);
                         if (property == null) {
-                            profilex = sessionService.fillProfileProperties(profilex, true);
+                            // Sapphire start - call GameProfileLookupEvent
+                            io.sapphiremc.sapphire.api.event.GameProfileLookupEvent event = new io.sapphiremc.sapphire.api.event.GameProfileLookupEvent(
+                                !org.bukkit.Bukkit.isPrimaryThread(), profilex.getId(), profilex.getName()
+                            );
+                            org.bukkit.Bukkit.getPluginManager().callEvent(event);
+                            GameProfile eventProfile = event.getGameProfile();
+                            if (eventProfile != null) {
+                                profilex = eventProfile;
+                            } else {
+                                profilex = sessionService.fillProfileProperties(profilex, true);
+                            }
+                            // Sapphire end
                         }
 
                         GameProfile gameProfile = profilex;
