From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 3 Dec 2021 19:45:00 +0500
Subject: [PATCH] Sapphire configuration


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index d73b7d72d00fa0edd6542226348af14c5203cf8d..689a579eba6100de0d432918ffbbb800b829deb1 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1949,6 +1949,14 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Purpur end
 
+        // Sapphire start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getSapphireConfig()
+        {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Sapphire end
+
         /**
          * Sends the component to the player
          *
