From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 3 Dec 2021 19:45:00 +0500
Subject: [PATCH] Sapphire configuration


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 730caafaf9629664ba125f3884559da76616f655..5221a7148d61d0b57a0e2950b96e5fccd003b3c4 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1980,6 +1980,14 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Purpur end
 
+        // Sapphire start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getSapphireConfig()
+        {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Sapphire end
+
         /**
          * Sends the component to the player
          *
