From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 3 Dec 2021 19:45:00 +0500
Subject: [PATCH] Sapphire configuration


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index d87e4d177b60957eee198c3f1791dbf26dbdddd6..53fa5a67f7bf58e13d5f3d715cfd7f22f545d241 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1979,6 +1979,14 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Purpur end
 
+        // Sapphire start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getSapphireConfig()
+        {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Sapphire end
+
         /**
          * Sends the component to the player
          *
