From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 3 Dec 2021 19:45:00 +0500
Subject: [PATCH] Sapphire configuration


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 15a0ae9744fa841e3e51e9e9a957cea8c71e68ad..ab5d055fef834fb0287ad947314e036b91fde395 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2019,6 +2019,14 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Purpur end
 
+        // Sapphire start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getSapphireConfig()
+        {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Sapphire end
+
         /**
          * Sends the component to the player
          *
