From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 3 Dec 2021 21:18:54 +0500
Subject: [PATCH] Add GameProfileLookupEvent


diff --git a/build.gradle.kts b/build.gradle.kts
index a141e5cdcec4d09b05252782e8a0ab5fac532680..82b4f76e0c75677f06e4d360b70e5309c9df3e0f 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -42,6 +42,7 @@ dependencies {
     api("org.apache.logging.log4j:log4j-api:2.17.1")
     api("org.slf4j:slf4j-api:1.8.0-beta4")
     api("io.sentry:sentry:5.4.0") // Pufferfish
+    api("com.mojang:authlib:2.3.31") // Sapphire
 
     implementation("org.ow2.asm:asm:9.2")
     implementation("org.ow2.asm:asm-commons:9.2")
diff --git a/src/main/java/io/sapphiremc/sapphire/api/event/GameProfileLookupEvent.java b/src/main/java/io/sapphiremc/sapphire/api/event/GameProfileLookupEvent.java
new file mode 100644
index 0000000000000000000000000000000000000000..b4e52e5f4e5218c1faaa141ac576670df1d93711
--- /dev/null
+++ b/src/main/java/io/sapphiremc/sapphire/api/event/GameProfileLookupEvent.java
@@ -0,0 +1,53 @@
+package io.sapphiremc.sapphire.api.event;
+
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import com.mojang.authlib.GameProfile;
+
+import java.util.UUID;
+
+public class GameProfileLookupEvent extends Event {
+
+    private static final HandlerList handlers = new HandlerList();
+    private GameProfile gameProfile = null;
+    private final UUID uuid;
+    private final String name;
+
+    public GameProfileLookupEvent(boolean async, @NotNull UUID uuid, @NotNull String name) {
+        super(async);
+        this.uuid = uuid;
+        this.name = name;
+    }
+
+    @Nullable
+    public GameProfile getGameProfile() {
+        return gameProfile;
+    }
+
+    public void setGameProfile(@Nullable GameProfile gameProfile) {
+        this.gameProfile = gameProfile;
+    }
+
+    @NotNull
+    public UUID getUuid() {
+        return uuid;
+    }
+
+    @NotNull
+    public String getName() {
+        return name;
+    }
+
+    @NotNull
+    @Override
+    public HandlerList getHandlers() {
+        return handlers;
+    }
+
+    @NotNull
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+}
