From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 11 Feb 2022 00:39:20 +0500
Subject: [PATCH] Chromium client support


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 5ff9f0d9bc2bc08af251a677a8c98d6f288da8a8..980b00edffdda72d919ee833f7c65499a9d7ef5a 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -3113,4 +3113,13 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
      */
     void sendDeathScreen(@NotNull Component message, @Nullable Entity killer);
     // Purpur end
+
+    // Sapphire start
+    /**
+     * Allows you to get if player uses Chromium Client
+     *
+     * @return True if Player uses Chromium Client
+     */
+    boolean usesChromiumClient();
+    // Sapphire end
 }
