From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 11 Feb 2022 00:39:20 +0500
Subject: [PATCH] Chromium client support


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 47fcfa2a3358766dfda2efc9bbcf5b50e3f2f7c1..af2aab3a2ef0807dbc5d863e84723719450e1b9b 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -3366,4 +3366,13 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     @Override
     Spigot spigot();
     // Spigot end
+
+    // Sapphire start
+    /**
+     * Allows you to get if player uses Chromium Client
+     *
+     * @return True if Player uses Chromium Client
+     */
+    boolean usesChromiumClient();
+    // Sapphire end
 }
