From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 11 Feb 2022 00:39:20 +0500
Subject: [PATCH] Chromium client support


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 9240ea09206461d61cc08d4252e8507555bf41cf..66d477ba99c622ed5786dbe3b8f2880602cf1a79 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -3307,4 +3307,13 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     @Override
     Spigot spigot();
     // Spigot end
+
+    // Sapphire start
+    /**
+     * Allows you to get if player uses Chromium Client
+     *
+     * @return True if Player uses Chromium Client
+     */
+    boolean usesChromiumClient();
+    // Sapphire end
 }
