From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 3 Dec 2021 20:19:22 +0500
Subject: [PATCH] Fix Javadoc generation


diff --git a/build.gradle.kts b/build.gradle.kts
index a23b2bd8e1ca1ff8d0ad5ed5d5e41c89e4795090..4c84a95366f7bed806e7e206cfa86fbe7b28a66f 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -102,6 +102,8 @@ tasks.withType<Javadoc> {
             apiAndDocs.resolvedConfiguration.files.joinToString(separator = File.pathSeparator, transform = File::getPath)
         )
     }
+    (options as StandardJavadocDocletOptions).encoding = "UTF-8"
+    (options as StandardJavadocDocletOptions).addBooleanOption("html5", true)
     (options as StandardJavadocDocletOptions).links(
         "https://guava.dev/releases/31.0.1-jre/api/docs/",
         "https://javadoc.io/doc/org.yaml/snakeyaml/1.30/",
