From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Sat, 4 Dec 2021 00:02:38 +0500
Subject: [PATCH] Add last tick time api


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index a3e22c90f856ad215bab4e0f308e46cd4c4b06db..184271c04738073fc46393e0f261b5cbec1f703d 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2565,4 +2565,27 @@ public final class Bukkit {
         server.clearBlockHighlights();
     }
     // Purpur end
+
+    // Sapphire start
+    /**
+     * Returns the time of the last tick took in milliseconds.
+     *
+     * @return long time value
+     * @deprecated never method with java's Duration and a possibility to convert it from millis to something else
+     */
+    @Deprecated
+    public static long getLastTickMs() {
+        return server.getLastTickMs();
+    }
+
+    /**
+     * Returns the time of the last tick took in {@link java.time.Duration}
+     *
+     * @return duration time value
+     */
+    @NotNull
+    public static java.time.Duration getLastTickDuration() {
+        return server.getLastTickDuration();
+    }
+    // Sapphire end
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index ed80d5553a1d4a6496b5513d2bc264ea817121d3..8935b9ae4c683f45fbdc4f830f913e756510c9b7 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2237,4 +2237,25 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
      */
     void clearBlockHighlights();
     // Purpur end
+
+    // Sapphire start
+    /**
+     * Returns the time of the last tick took in milliseconds.
+     *
+     * @return long time value
+     * @deprecated never method with java's Duration and a possibility to convert it from millis to something else
+     */
+    @Deprecated
+    default long getLastTickMs() {
+        return getLastTickDuration().toMillis();
+    }
+
+    /**
+     * Returns the time of the last tick took in {@link java.time.Duration}
+     *
+     * @return duration time value
+     */
+    @NotNull
+    java.time.Duration getLastTickDuration();
+    // Sapphire end
 }
