From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Sat, 4 Dec 2021 00:02:38 +0500
Subject: [PATCH] Add last tick time api


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 9b9af189b3865055570ba533b97c92c105cc5a37..f89f8c1e5272d0cbcc6489c4d5a7b6b1e8b430d3 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2468,4 +2468,27 @@ public final class Bukkit {
         server.removeFuel(material);
     }
     // Purpur end
+
+    // Sapphire start
+    /**
+     * Returns the time of the last tick took in milliseconds.
+     *
+     * @return long time value
+     * @deprecated never method with java's Duration and a possibility to convert it from millis to something else
+     */
+    @Deprecated
+    public static long getLastTickMs() {
+        return server.getLastTickMs();
+    }
+
+    /**
+     * Returns the time of the last tick took in {@link java.time.Duration}
+     *
+     * @return duration time value
+     */
+    @NotNull
+    public static java.time.Duration getLastTickDuration() {
+        return server.getLastTickDuration();
+    }
+    // Sapphire end
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 689a579eba6100de0d432918ffbbb800b829deb1..29fc0c39f241362fda86a7d7a790c189ff9693d8 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2156,4 +2156,25 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
      */
     public void removeFuel(@NotNull Material material);
     // Purpur end
+
+    // Sapphire start
+    /**
+     * Returns the time of the last tick took in milliseconds.
+     *
+     * @return long time value
+     * @deprecated never method with java's Duration and a possibility to convert it from millis to something else
+     */
+    @Deprecated
+    default long getLastTickMs() {
+        return getLastTickDuration().toMillis();
+    }
+
+    /**
+     * Returns the time of the last tick took in {@link java.time.Duration}
+     *
+     * @return duration time value
+     */
+    @NotNull
+    java.time.Duration getLastTickDuration();
+    // Sapphire end
 }
