From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Sat, 4 Dec 2021 00:02:38 +0500
Subject: [PATCH] Add last tick time api


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index fc16e897f4ebaf9d79e49774e15a1cded33ef725..ff70e4a8e6c68c40c257548758fb33a5659c0701 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2381,4 +2381,27 @@ public final class Bukkit {
     public static Server.Spigot spigot() {
         return server.spigot();
     }
+
+    // Sapphire start
+    /**
+     * Returns the time of the last tick took in milliseconds.
+     *
+     * @return long time value
+     * @deprecated never method with java's Duration and a possibility to convert it from millis to something else
+     */
+    @Deprecated
+    public static long getLastTickMs() {
+        return server.getLastTickMs();
+    }
+
+    /**
+     * Returns the time of the last tick took in {@link java.time.Duration}
+     *
+     * @return duration time value
+     */
+    @NotNull
+    public static java.time.Duration getLastTickDuration() {
+        return server.getLastTickDuration();
+    }
+    // Sapphire end
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index c42a2e1e76aca93d2d47a4a9ec451efb45cb05a7..aa590bd0ef356984703ca0304f205ad65d727793 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2073,4 +2073,25 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
      */
     @NotNull org.bukkit.potion.PotionBrewer getPotionBrewer();
     // Paper end
+
+    // Sapphire start
+    /**
+     * Returns the time of the last tick took in milliseconds.
+     *
+     * @return long time value
+     * @deprecated never method with java's Duration and a possibility to convert it from millis to something else
+     */
+    @Deprecated
+    default long getLastTickMs() {
+        return getLastTickDuration().toMillis();
+    }
+
+    /**
+     * Returns the time of the last tick took in {@link java.time.Duration}
+     *
+     * @return duration time value
+     */
+    @NotNull
+    java.time.Duration getLastTickDuration();
+    // Sapphire end
 }
