From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Sat, 4 Dec 2021 00:02:38 +0500
Subject: [PATCH] Add last tick time api


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 944f9b87a11472ac6d7e328acc00bf09f899e648..4ca3629b1bafc339eba2c791711ccc6f3e71e536 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2312,4 +2312,27 @@ public final class Bukkit {
     public static Server.Spigot spigot() {
         return server.spigot();
     }
+
+    // Sapphire start
+    /**
+     * Returns the time of the last tick took in milliseconds.
+     *
+     * @return long time value
+     * @deprecated never method with java's Duration and a possibility to convert it from millis to something else
+     */
+    @Deprecated
+    public static long getLastTickMs() {
+        return server.getLastTickMs();
+    }
+
+    /**
+     * Returns the time of the last tick took in {@link java.time.Duration}
+     *
+     * @return duration time value
+     */
+    @NotNull
+    public static java.time.Duration getLastTickTime() {
+        return server.getLastTickTime();
+    }
+    // Sapphire end
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 239a9666bf404d38641883b2157e6c199f7de0c1..db20f6229e01f85b95066071399ed52f493e8b51 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2011,4 +2011,25 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
      */
     @NotNull org.bukkit.potion.PotionBrewer getPotionBrewer();
     // Paper end
+
+    // Sapphire start
+    /**
+     * Returns the time of the last tick took in milliseconds.
+     *
+     * @return long time value
+     * @deprecated never method with java's Duration and a possibility to convert it from millis to something else
+     */
+    @Deprecated
+    default long getLastTickMs() {
+        return getLastTickTime().toMillis();
+    }
+
+    /**
+     * Returns the time of the last tick took in {@link java.time.Duration}
+     *
+     * @return duration time value
+     */
+    @NotNull
+    java.time.Duration getLastTickTime();
+    // Sapphire end
 }
