From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 11 Feb 2022 00:39:20 +0500
Subject: [PATCH] Chromium client support


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 6cd0b10d1dc4506cfb1e4db5e1260cb705566cec..cbbdc94e1e71550dde955b76227d379fc3f73399 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -2694,6 +2694,15 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     String getClientBrandName();
     // Paper end
 
+    // Sapphire start - Chromium client support
+    /**
+     * Allows you to get if player uses Chromium Client
+     *
+     * @return True if Player uses Chromium Client
+     */
+    boolean usesChromiumClient();
+    // Sapphire end
+
     @NotNull
     @Override
     Spigot spigot();
