From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Sat, 4 Dec 2021 00:02:38 +0500
Subject: [PATCH] Add last tick time api


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 3a8326735b62521f8fb95c51a0909d8b7bac83d1..080378586b51e333f379fd8137a78478731f96a9 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2195,4 +2195,27 @@ public final class Bukkit {
     public static Server.Spigot spigot() {
         return server.spigot();
     }
+
+    // Sapphire start
+    /**
+     * Returns the time of the last tick took in milliseconds.
+     *
+     * @return long time value
+     * @deprecated never method with java's Duration and a possibility to convert it from millis to something else
+     */
+    @Deprecated
+    public static long getLastTickMs() {
+        return server.getLastTickMs();
+    }
+
+    /**
+     * Returns the time of the last tick took in {@link java.time.Duration}
+     *
+     * @return duration time value
+     */
+    @NotNull
+    public static java.time.Duration getLastTickTime() {
+        return server.getLastTickTime();
+    }
+    // Sapphire end
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 68e8ee865c5948905320c7886fb7b3e83a43d073..a3f6090508ab2452a9a72b099b20e614afade506 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1908,4 +1908,25 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
     @NotNull
     io.papermc.paper.datapack.DatapackManager getDatapackManager();
     // Paper end
+
+    // Sapphire start
+    /**
+     * Returns the time of the last tick took in milliseconds.
+     *
+     * @return long time value
+     * @deprecated never method with java's Duration and a possibility to convert it from millis to something else
+     */
+    @Deprecated
+    default long getLastTickMs() {
+        return getLastTickTime().toMillis();
+    }
+
+    /**
+     * Returns the time of the last tick took in {@link java.time.Duration}
+     *
+     * @return duration time value
+     */
+    @NotNull
+    java.time.Duration getLastTickTime();
+    // Sapphire end
 }
