From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 11 Feb 2022 00:39:20 +0500
Subject: [PATCH] Chromium client support


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 8ca135d2e3682581855399e7c6d07a2d13244600..6da93a3e4ead19ef483cbf213c6c6e4b6629340d 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -2692,6 +2692,15 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     String getClientBrandName();
     // Paper end
 
+    // Sapphire start - Chromium client support
+    /**
+     * Allows you to get if player uses Chromium Client
+     *
+     * @return True if Player uses Chromium Client
+     */
+    boolean usesChromiumClient();
+    // Sapphire end
+
     @NotNull
     @Override
     Spigot spigot();
