From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Mon, 2 May 2022 01:17:41 +0500
Subject: [PATCH] Add Player.getFishingHook API

Ported from EmpireCraft

diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index e7642be6723849e00989637bb0385505c46b1f00..d251eb1f168b77f580b8e345da7768125f979b05 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -2799,6 +2799,18 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
      * @return True if Player uses Chromium Client
      */
     boolean usesChromiumClient();
+
+    /**
+     * Gets the current {@link FishHook} a player has attached.
+     *
+     * @return current fisging hook or null if the player is not fishing
+     * @deprecated use {@link HumanEntity#getFishHook()}
+     */
+    @Nullable
+    @Deprecated
+    default FishHook getFishingHook() {
+        return getFishHook();
+    }
     // Sapphire end
 
     @NotNull
