From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Mon, 2 May 2022 01:17:41 +0500
Subject: [PATCH] Add Player.getFishingHook API

Ported from EmpireCraft

diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 78b9cbb0049f696b69c7d1700f4d28b579e79bf0..fe0d655cd62b6787edcb7361b67b85eda5a8ee69 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -2843,6 +2843,18 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
      * @return True if Player uses Chromium Client
      */
     boolean usesChromiumClient();
+
+    /**
+     * Gets the current {@link FishHook} a player has attached.
+     *
+     * @return current fisging hook or null if the player is not fishing
+     * @deprecated use {@link HumanEntity#getFishHook()}
+     */
+    @Nullable
+    @Deprecated
+    default FishHook getFishingHook() {
+        return getFishHook();
+    }
     // Sapphire end
 
     @NotNull
