From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Mon, 2 May 2022 01:17:41 +0500
Subject: [PATCH] Add Player.getFishingHook API

Ported from EmpireCraft

diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 02b59ab102ccf59f9c420e80c06fe9db10140a76..9138c69f8ae7d0b982ccfe7914c69e448f2caa79 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -2773,6 +2773,18 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
      * @return True if Player uses Chromium Client
      */
     boolean usesChromiumClient();
+
+    /**
+     * Gets the current {@link FishHook} a player has attached.
+     *
+     * @return current fisging hook or null if the player is not fishing
+     * @deprecated use {@link HumanEntity#getFishHook()}
+     */
+    @Nullable
+    @Deprecated
+    default FishHook getFishingHook() {
+        return getFishHook();
+    }
     // Sapphire end
 
     @NotNull
