From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Mon, 2 May 2022 01:17:41 +0500
Subject: [PATCH] Add Player.getFishingHook API

Ported from EmpireCraft

diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index cbbdc94e1e71550dde955b76227d379fc3f73399..c618e7e6ae001ea7ff8704dec18d31a90979e972 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -2694,13 +2694,21 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     String getClientBrandName();
     // Paper end
 
-    // Sapphire start - Chromium client support
+    // Sapphire start
     /**
      * Allows you to get if player uses Chromium Client
      *
      * @return True if Player uses Chromium Client
      */
     boolean usesChromiumClient();
+
+    /**
+     * Gets the current {@link FishHook} a player has attached.
+     *
+     * @return current fisging hook or null if the player is not fishing
+     */
+    @Nullable
+    FishHook getFishingHook();
     // Sapphire end
 
     @NotNull
