From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 11 Feb 2022 00:39:20 +0500
Subject: [PATCH] Sapphire client support


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 131daee2b29f7016463a00ce7927dff7b0a1b1b4..20d58570504bf1adaca20897e89733c5ea1b59bb 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -2648,6 +2648,15 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     String getClientBrandName();
     // Paper end
 
+    // Sapphire start - our client support
+    /**
+     * Allows you to get if player uses Sapphire Client
+     *
+     * @return True if Player uses Sapphire Client
+     */
+    boolean usesSapphireClient();
+    // Sapphire end
+
     @NotNull
     @Override
     Spigot spigot();
