From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Mon, 6 Jun 2022 00:15:47 +0500
Subject: [PATCH] Skip event if no listeners


diff --git a/src/main/java/org/bukkit/plugin/SimplePluginManager.java b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
index 6bb115923767c4bd65b18e67eeff5408f2894ab0..4d8d9e214829552f35d2e7ac58b0fae5e7a99a01 100644
--- a/src/main/java/org/bukkit/plugin/SimplePluginManager.java
+++ b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
@@ -659,6 +659,12 @@ public final class SimplePluginManager implements PluginManager {
      */
     @Override
     public void callEvent(@NotNull Event event) {
+        // Sapphire start
+        HandlerList handlers = event.getHandlers();
+        RegisteredListener[] listeners = handlers.getRegisteredListeners();
+        if (listeners.length == 0) return;
+        // Sapphire end
+
         // Paper - replace callEvent by merging to below method
         if (event.isAsynchronous() && server.isPrimaryThread()) {
             throw new IllegalStateException(event.getEventName() + " may only be triggered asynchronously.");
@@ -666,8 +672,8 @@ public final class SimplePluginManager implements PluginManager {
             throw new IllegalStateException(event.getEventName() + " may only be triggered synchronously.");
         }
 
-        HandlerList handlers = event.getHandlers();
-        RegisteredListener[] listeners = handlers.getRegisteredListeners();
+       // HandlerList handlers = event.getHandlers(); // Sapphire
+       // RegisteredListener[] listeners = handlers.getRegisteredListeners(); // Sapphire
 
         for (RegisteredListener registration : listeners) {
             if (!registration.getPlugin().isEnabled()) {
