From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Mon, 6 Jun 2022 00:15:47 +0500
Subject: [PATCH] Skip event if no listeners

Ported from JettPack

diff --git a/src/main/java/org/bukkit/plugin/SimplePluginManager.java b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
index ed07f5820281b139739f673fa4e25171de81b894..07edef636946a5e8707689fa3ac32b985bdad1ad 100644
--- a/src/main/java/org/bukkit/plugin/SimplePluginManager.java
+++ b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
@@ -645,6 +645,12 @@ public final class SimplePluginManager implements PluginManager {
      */
     @Override
     public void callEvent(@NotNull Event event) {
+        // Sapphire start
+        HandlerList handlers = event.getHandlers();
+        RegisteredListener[] listeners = handlers.getRegisteredListeners();
+        if (listeners.length == 0) return;
+        // Sapphire end
+
         // Paper - replace callEvent by merging to below method
         if (event.isAsynchronous() && server.isPrimaryThread()) {
             throw new IllegalStateException(event.getEventName() + " may only be triggered asynchronously.");
@@ -652,8 +658,8 @@ public final class SimplePluginManager implements PluginManager {
             throw new IllegalStateException(event.getEventName() + " may only be triggered synchronously.");
         }
 
-        HandlerList handlers = event.getHandlers();
-        RegisteredListener[] listeners = handlers.getRegisteredListeners();
+       // HandlerList handlers = event.getHandlers(); // Sapphire
+       // RegisteredListener[] listeners = handlers.getRegisteredListeners(); // Sapphire
 
         for (RegisteredListener registration : listeners) {
             if (!registration.getPlugin().isEnabled()) {
