From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Sat, 4 Dec 2021 00:08:30 +0500
Subject: [PATCH] Add item stuck sleep config


diff --git a/src/main/java/io/sapphiremc/sapphire/SapphireConfig.java b/src/main/java/io/sapphiremc/sapphire/SapphireConfig.java
index 795e5b12ec00187e420025c05ee3b979838abd10..a8cf5e06636f047c9d54fd9f7567434bc7b3c418 100644
--- a/src/main/java/io/sapphiremc/sapphire/SapphireConfig.java
+++ b/src/main/java/io/sapphiremc/sapphire/SapphireConfig.java
@@ -127,8 +127,10 @@ public class SapphireConfig {
     }
 
     public static boolean disableEntityStuckChecks = false;
+    public static int itemStuckSleepTicks = 1;
     private static void settings() {
         disableEntityStuckChecks = getBoolean("settings.disable-entity-stuck-checks", false);
+        itemStuckSleepTicks = getInt("settings.item-stuck-sleep-ticks", itemStuckSleepTicks);
     }
 
     public static boolean playerCollisions = true;
diff --git a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
index 52a14af24e3c51c53b40fdb6594b1664eef6d486..22e0cc38893c02ff37fbae4704be0089837b1da4 100644
--- a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
+++ b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
@@ -155,7 +155,7 @@ public class ItemEntity extends Entity implements TraceableEntity {
 
             if (this.level.isClientSide) {
                 this.noPhysics = false;
-            } else {
+            } else if (!this.onGround || this.noPhysics || this.age * io.sapphiremc.sapphire.SapphireConfig.itemStuckSleepTicks == 0) { // Sapphire
                 this.noPhysics = !this.level.noCollision(this, this.getBoundingBox().deflate(1.0E-7D));
                 if (this.noPhysics) {
                     this.moveTowardsClosestSpace(this.getX(), (this.getBoundingBox().minY + this.getBoundingBox().maxY) / 2.0D, this.getZ());
