From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Sat, 4 Dec 2021 00:08:30 +0500
Subject: [PATCH] Add item stuck sleep config


diff --git a/src/main/java/io/sapphiremc/sapphire/SapphireConfig.java b/src/main/java/io/sapphiremc/sapphire/SapphireConfig.java
index 38deac094eccb6eb6c2305a01f3760e98fc721e4..f124203e74fa6df2d183078d5eb6e4f2e1f45c0b 100644
--- a/src/main/java/io/sapphiremc/sapphire/SapphireConfig.java
+++ b/src/main/java/io/sapphiremc/sapphire/SapphireConfig.java
@@ -128,9 +128,11 @@ public class SapphireConfig {
 
     public static String serverModName = "Sapphire";
     public static boolean disableEntityStuckChecks = false;
+    public static int itemStuckSleepTicks = 1;
     private static void settings() {
         serverModName = getString("settings.server-mod-name", serverModName);
         disableEntityStuckChecks = getBoolean("settings.disable-entity-stuck-checks", disableEntityStuckChecks);
+        itemStuckSleepTicks = getInt("settings.item-stuck-sleep-ticks", itemStuckSleepTicks);
     }
 
     public static boolean playerCollisions = true;
diff --git a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
index 80fc7918cb294b0d88a293bd6a920441cb55c3ad..16b07d2bddf65d81cd03804ac1f2554b925387f6 100644
--- a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
+++ b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
@@ -150,7 +150,7 @@ public class ItemEntity extends Entity implements TraceableEntity {
 
             if (this.level().isClientSide) {
                 this.noPhysics = false;
-            } else {
+            } else if (!this.onGround || this.noPhysics || this.age * io.sapphiremc.sapphire.SapphireConfig.itemStuckSleepTicks == 0) { // Sapphire
                 this.noPhysics = !this.level().noCollision(this, this.getBoundingBox().deflate(1.0E-7D));
                 if (this.noPhysics) {
                     this.moveTowardsClosestSpace(this.getX(), (this.getBoundingBox().minY + this.getBoundingBox().maxY) / 2.0D, this.getZ());
