From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 31 Dec 2021 02:03:55 +0500
Subject: [PATCH] Global eula file


diff --git a/src/main/java/net/minecraft/server/Eula.java b/src/main/java/net/minecraft/server/Eula.java
index 4996694ec43a894cbbb363e48bfdff6b6ae17b5f..f948b8531f8a397dac9379940ca25dbefc5da0b8 100644
--- a/src/main/java/net/minecraft/server/Eula.java
+++ b/src/main/java/net/minecraft/server/Eula.java
@@ -16,16 +16,28 @@ public class Eula {
 
     public Eula(Path eulaFile) {
         this.file = eulaFile;
-        this.agreed = SharedConstants.IS_RUNNING_IN_IDE || this.readFile();
+        this.agreed = SharedConstants.IS_RUNNING_IN_IDE || globalEula() || this.readFile(this.file);
     }
 
-    private boolean readFile() {
-        try (InputStream inputStream = Files.newInputStream(this.file)) {
+    // Sapphire start - Global eula file
+    private boolean globalEula() {
+        java.io.File globalEula = new java.io.File(System.getProperty("user.home"), "eula.txt");
+        if (globalEula.exists()) {
+            return this.readFile(globalEula.toPath());
+        } else {
+            LOGGER.info("No global eula found at " + globalEula.getAbsolutePath());
+        }
+        return false;
+    }
+    // Sapphire end
+
+    private boolean readFile(java.nio.file.Path path) { // Sapphire - Global eula file
+        try (InputStream inputStream = Files.newInputStream(path)) { // Sapphire - Global eula file
             Properties properties = new Properties();
             properties.load(inputStream);
             return Boolean.parseBoolean(properties.getProperty("eula", "false"));
         } catch (Exception var6) {
-            LOGGER.warn("Failed to load {}", (Object)this.file);
+            LOGGER.warn("Failed to load {}", (Object)path); // Sapphire - Global eula file
             this.saveDefaults();
             return false;
         }
