From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Sat, 4 Dec 2021 00:08:30 +0500
Subject: [PATCH] Add item stuck sleep config


diff --git a/src/main/java/io/sapphiremc/sapphire/SapphireConfig.java b/src/main/java/io/sapphiremc/sapphire/SapphireConfig.java
index ec6af9df782e317fdde35ae9bda73ffdb11e73fd..1e70892e4fd9e7401ea269ba6c9eae1facab708f 100644
--- a/src/main/java/io/sapphiremc/sapphire/SapphireConfig.java
+++ b/src/main/java/io/sapphiremc/sapphire/SapphireConfig.java
@@ -169,8 +169,10 @@ public class SapphireConfig {
     }
 
     public static boolean disableEntityStuckChecks = false;
+    public static int itemStuckSleepTicks = 1;
     private static void settings() {
         disableEntityStuckChecks = getBoolean("settings.disable-entity-stuck-checks", false);
+        itemStuckSleepTicks = getInt("settings.item-stuck-sleep-ticks", itemStuckSleepTicks);
     }
 
     public static boolean playerCollisions = true;
diff --git a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
index 7e927dec3e4e4fb65bf89a87cfa555045c58bb3a..4f0034a3375d49b7e2c71ae201dcc94b13370cb1 100644
--- a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
+++ b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
@@ -138,7 +138,7 @@ public class ItemEntity extends Entity {
 
             if (this.level.isClientSide) {
                 this.noPhysics = false;
-            } else {
+            } else if (!this.onGround || this.noPhysics || this.age * io.sapphiremc.sapphire.SapphireConfig.itemStuckSleepTicks == 0) { // Sapphire
                 this.noPhysics = !this.level.noCollision(this, this.getBoundingBox().deflate(1.0E-7D));
                 if (this.noPhysics) {
                     this.moveTowardsClosestSpace(this.getX(), (this.getBoundingBox().minY + this.getBoundingBox().maxY) / 2.0D, this.getZ());
