From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 31 Dec 2021 02:03:55 +0500
Subject: [PATCH] Global eula file


diff --git a/src/main/java/net/minecraft/server/Eula.java b/src/main/java/net/minecraft/server/Eula.java
index 5528a4b7a67e0c257966c587437006004a2e1292..8075290752dc35fa0d44aefa93efa0a2e6c0f13b 100644
--- a/src/main/java/net/minecraft/server/Eula.java
+++ b/src/main/java/net/minecraft/server/Eula.java
@@ -16,13 +16,25 @@ public class Eula {
 
     public Eula(Path eulaFile) {
         this.file = eulaFile;
-        this.agreed = SharedConstants.IS_RUNNING_IN_IDE || this.readFile();
+        this.agreed = SharedConstants.IS_RUNNING_IN_IDE || globalEula() || this.readFile(this.file);
     }
 
-    private boolean readFile() {
+    // Sapphire start - Global eula file
+    private boolean globalEula() {
+        java.io.File globalEula = new java.io.File(System.getProperty("user.home"), "eula.txt");
+        if (globalEula.exists()) {
+            return this.readFile(globalEula.toPath());
+        } else {
+            LOGGER.info("No global eula found at " + globalEula.getAbsolutePath());
+        }
+        return false;
+    }
+    // Sapphire end
+
+    private boolean readFile(java.nio.file.Path path) { // Sapphire - Global eula file
         try {
             boolean var3;
-            try (InputStream inputStream = Files.newInputStream(this.file)) {
+            try (InputStream inputStream = Files.newInputStream(path)) { // Sapphire - Global eula file
                 Properties properties = new Properties();
                 properties.load(inputStream);
                 var3 = Boolean.parseBoolean(properties.getProperty("eula", "false"));
@@ -30,7 +42,7 @@ public class Eula {
 
             return var3;
         } catch (Exception var6) {
-            LOGGER.warn("Failed to load {}", (Object)this.file);
+            LOGGER.warn("Failed to load {}", (Object)path); // Sapphire - Global eula file
             this.saveDefaults();
             return false;
         }
