From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 11 Feb 2022 00:38:44 +0500
Subject: [PATCH] Chromium client support


diff --git a/src/main/java/io/sapphiremc/sapphire/SapphireCommand.java b/src/main/java/io/sapphiremc/sapphire/SapphireCommand.java
index 770fdc9e52247b6fe0e23e8d9a5678ee66e4c68f..be44e3b3336eaa1e00f37e6916e7037d2e00c6b4 100644
--- a/src/main/java/io/sapphiremc/sapphire/SapphireCommand.java
+++ b/src/main/java/io/sapphiremc/sapphire/SapphireCommand.java
@@ -7,6 +7,7 @@ import org.bukkit.Bukkit;
 import org.bukkit.Location;
 import org.bukkit.command.Command;
 import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
@@ -63,6 +64,22 @@ public class SapphireCommand extends Command {
                     sender.sendMessage(prefix + "Sapphire config reload complete, but some functions need you to restart the server.");
                 }
             }
+            case "isChromium" -> {
+                if (args.length == 2) {
+                    Player player = Bukkit.getServer().getPlayer(args[1]);
+                    if (player != null) {
+                        if (player.usesChromiumClient()) {
+                            sender.sendMessage(prefix + "Player %s is using chromium client".formatted(args[1]));
+                        } else {
+                            sender.sendMessage(prefix + "Player %s isn't using chromium client".formatted(args[1]));
+                        }
+                    } else {
+                        sender.sendMessage(prefix + "Player %s is null".formatted(args[1]));
+                    }
+                } else {
+                    sender.sendMessage(prefix + "Usage: /sapphire isChromium <name>");
+                }
+            }
             case "ver", "version" -> {
                 if (args.length == 1) {
                     sender.sendMessage(prefix + "This server is running " + Bukkit.getName() + " version " + Bukkit.getVersion() + " (Git: " + getGitInfo() + ")");
@@ -74,7 +91,7 @@ public class SapphireCommand extends Command {
         return true;
     }
 
-    private final Collection<String> allAliases = Set.of("reload", "version", "ver");
+    private final Collection<String> allAliases = Set.of("reload", "isChromium", "version", "ver");
 
     @Override
     public @NotNull List<String> tabComplete(@NotNull CommandSender sender, @NotNull String alias, @NotNull String[] args, @Nullable Location location) throws IllegalArgumentException {
@@ -92,6 +109,20 @@ public class SapphireCommand extends Command {
                     .toList()
                 );
             }
+        } else if (args.length == 2) {
+            if (args[0].isEmpty()) {
+                aliases.addAll(Bukkit.getOnlinePlayers().stream()
+                    .map(Player::getName)
+                    .toList()
+                );
+            } else {
+                String arg = args[1];
+                aliases.addAll(Bukkit.getOnlinePlayers().stream()
+                    .map(Player::getName)
+                    .filter(s -> s.contains(arg))
+                    .toList()
+                );
+            }
         }
 
         return aliases;
diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
index 74238a87d1ff3391aac5812b24b84af228baeaa4..c7dbeeeb840f3e5062c50b257853dc45d2f7b4bf 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
@@ -279,6 +279,7 @@ public class ServerPlayer extends Player {
     private boolean ramBar = false; // Purpur
     private boolean tpsBar = false; // Purpur
     private boolean compassBar = false; // Purpur
+    public boolean chromiumClient = false; // Sapphire
 
     public ServerPlayer(MinecraftServer server, ServerLevel world, GameProfile profile) {
         super(world, world.getSharedSpawnPos(), world.getSharedSpawnAngle(), profile);
diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index ae3c1bd67c144fe971d4df6df2ca171b431b6fc5..307a819579153126de2bb53c365b32868fe3bb00 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -3642,6 +3642,7 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
 
     private static final ResourceLocation MINECRAFT_BRAND = new ResourceLocation("brand"); // Paper - Brand support
     private static final ResourceLocation PURPUR_CLIENT = new ResourceLocation("purpur", "client"); // Purpur
+    private static final ResourceLocation CHROMIUM_CLIENT = new ResourceLocation("chromium", "client"); // Sapphire
 
     @Override
     public void handleCustomPayload(ServerboundCustomPayloadPacket packet) {
@@ -3673,6 +3674,13 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
             } catch (Exception ignore) {
             }
         // Purpur end
+        // Sapphire start
+        } else if (packet.identifier.equals(CHROMIUM_CLIENT)) {
+            try {
+                player.chromiumClient = true;
+            } catch (Exception ignored) {
+            }
+        // Sapphire end
         } else {
             try {
                 byte[] data = new byte[packet.data.readableBytes()];
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 2b647aeb75e2207186bf9506fba0d92a745cfae8..a536424d79e7ef9437fc90f5b4257f862e089d2e 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -3272,4 +3272,11 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         this.getHandle().connection.send(packet);
     }
     // Purpur end
+
+    // Sapphire start
+    @Override
+    public boolean usesChromiumClient() {
+        return getHandle().chromiumClient;
+    }
+    // Sapphire end
 }
