From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Mon, 2 May 2022 01:27:17 +0500
Subject: [PATCH] Allow force spawns

Ported from EmpireCraft

diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index b396a329e753f42b3878736b6b14bafda02d3744..ef2977cb9bfa7a6070be6d62d2d2c863c9d136f1 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -1522,7 +1522,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
         // SPIGOT-6415: Don't call spawn event for entities which travel trough worlds,
         // since it is only an implementation detail, that a new entity is created when
         // they are traveling between worlds.
-        this.addDuringTeleport(entity, null);
+        this.addDuringTeleport(entity, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.FORCE); // Sapphire
     }
 
     public void addDuringTeleport(Entity entity, CreatureSpawnEvent.SpawnReason reason) {
diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index 2d49774f195c2f72004319b2e35763c7071969f1..54f4ad26be2695ee2c10e6bea912ba58d9e9ba99 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -713,7 +713,7 @@ public class CraftEventFactory {
             event = CraftEventFactory.callEntitySpawnEvent(entity);
         }
 
-        if (event != null && (event.isCancelled() || entity.isRemoved())) {
+        if (spawnReason != SpawnReason.FORCE && event != null && (event.isCancelled() || entity.isRemoved())) { // Sapphire
             Entity vehicle = entity.getVehicle();
             if (vehicle != null) {
                 vehicle.discard();
