From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Mon, 2 May 2022 01:27:17 +0500
Subject: [PATCH] Allow force spawns

Ported from EmpireCraft

diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index eac31c3fcc9161711328588ac852fcae1116d8ef..727cc7a5fd26ad081c1f53d48ea218b5a12d1da3 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -1596,7 +1596,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
         // SPIGOT-6415: Don't call spawn event for entities which travel trough worlds,
         // since it is only an implementation detail, that a new entity is created when
         // they are traveling between worlds.
-        this.addDuringTeleport(entity, null);
+        this.addDuringTeleport(entity, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.FORCE); // Sapphire
     }
 
     public void addDuringTeleport(Entity entity, CreatureSpawnEvent.SpawnReason reason) {
diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index acb38f927c2418c91adf91c7b790ea6869835537..bc39253a54855cfd1d05360622eac45f07e9bb18 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -727,7 +727,7 @@ public class CraftEventFactory {
             event = CraftEventFactory.callEntitySpawnEvent(entity);
         }
 
-        if (event != null && (event.isCancelled() || entity.isRemoved())) {
+        if (spawnReason != SpawnReason.FORCE && event != null && (event.isCancelled() || entity.isRemoved())) { // Sapphire
             Entity vehicle = entity.getVehicle();
             if (vehicle != null) {
                 vehicle.discard();
