From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Fri, 11 Feb 2022 00:38:44 +0500
Subject: [PATCH] Chromium client support


diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
index 1013618b91df8b2bdf2aa3f5f9f91f0af5a8a7f0..d2a70c7a25ba7ff2003e6158ea0470dce6549b75 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
@@ -266,6 +266,7 @@ public class ServerPlayer extends Player {
     public boolean acceptingResourcePack = false; // Purpur
     private boolean tpsBar = false; // Purpur
     private boolean compassBar = false; // Purpur
+    public boolean chromiumClient = false; // Sapphire
 
     public double lastEntitySpawnRadiusSquared; // Paper - optimise isOutsideRange, this field is in blocks
     public final com.destroystokyo.paper.util.misc.PooledLinkedHashSets.PooledObjectLinkedOpenHashSet<ServerPlayer> cachedSingleHashSet; // Paper
diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 322bb76994dc3bf3eb21ac6c41c6d744743c766f..1deca736062d44ffa5c8b9c6988597ce6b44a95c 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -3704,6 +3704,8 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
     private static final ResourceLocation MINECRAFT_BRAND = new ResourceLocation("brand"); // Paper - Brand support
     private static final ResourceLocation PURPUR_CLIENT = new ResourceLocation("purpur", "client"); // Purpur
 
+    private static final ResourceLocation CHROMIUM_CLIENT = new ResourceLocation("sapphiremc", "chromium"); // Sapphire - Chromium client support
+
     @Override
     public void handleCustomPayload(ServerboundCustomPayloadPacket packet) {
         PacketUtils.ensureRunningOnSameThread(packet, this, this.player.getLevel());
@@ -3734,6 +3736,13 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
             } catch (Exception ignore) {
             }
         // Purpur end
+        // Sapphire start - Chromium client support
+        } else if (packet.identifier.equals(CHROMIUM_CLIENT)) {
+            try {
+                player.chromiumClient = true;
+            } catch (Exception ignored) {
+            }
+        // Sapphire end
         } else {
             try {
                 byte[] data = new byte[packet.data.readableBytes()];
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 753ef343b7daab339a4308935e3cec86e49481f0..f7d22d565f5d978b987d6d1e8c9e94787aa1add9 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2964,4 +2964,11 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         return getHandle().flyingFallDamage;
     }
     // Purpur end
+
+    // Sapphire start
+    @Override
+    public boolean usesChromiumClient() {
+        return getHandle().chromiumClient;
+    }
+    // Sapphire end
 }
