From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Sun, 18 Jun 2023 01:00:42 +0500
Subject: [PATCH] Include kotlin-stdlib in server jar


diff --git a/build.gradle.kts b/build.gradle.kts
index 316663e83d09f15024de586651c37064b5752a71..761653acaebbc0788eb58051d6b9f6bcd50e3f84 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -56,6 +56,9 @@ dependencies {
     runtimeOnly("org.apache.maven.resolver:maven-resolver-connector-basic:1.7.3")
     runtimeOnly("org.apache.maven.resolver:maven-resolver-transport-http:1.7.3")
 
+    implementation("org.jetbrains.kotlin:kotlin-stdlib:1.9.10") // Sapphire
+    implementation("org.jetbrains.kotlin:kotlin-reflect:1.9.10") // Sapphire
+
     // Pufferfish start
     implementation("org.yaml:snakeyaml:1.32")
     implementation ("me.carleslc.Simple-YAML:Simple-Yaml:1.8.4") {
