From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Mon, 2 May 2022 01:17:40 +0500
Subject: [PATCH] Add Player.getFishingHook API

Ported from EmpireCraft

diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 6e50f76dc6a73ef6d0e2f0aa57bfea464fabe1f9..b65be54793329319d2151c2c2947f5d213a746c8 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2784,11 +2784,19 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
     }
     // Paper end
 
-    // Sapphire start - Chromium client support
+    // Sapphire start
     @Override
     public boolean usesChromiumClient() {
         return getHandle().chromiumClient;
     }
+
+    @Override
+    public org.bukkit.entity.FishHook getFishingHook() {
+        if (getHandle().fishing != null) {
+            return (org.bukkit.entity.FishHook) getHandle().fishing.getBukkitEntity();
+        }
+        return null;
+    }
     // Sapphire end
 
     public Player.Spigot spigot()
