From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Sun, 18 Jun 2023 01:00:42 +0500
Subject: [PATCH] Include kotlin-stdlib in server jar


diff --git a/build.gradle.kts b/build.gradle.kts
index c964dada7618d1bec492ecc64121caf2ba23c248..f7913948e5794f40542c7be8f42dca982b60255f 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -55,6 +55,9 @@ dependencies {
     runtimeOnly("org.apache.maven.resolver:maven-resolver-connector-basic:1.7.3")
     runtimeOnly("org.apache.maven.resolver:maven-resolver-transport-http:1.7.3")
 
+    implementation("org.jetbrains.kotlin:kotlin-stdlib:1.9.22") // Sapphire
+    implementation("org.jetbrains.kotlin:kotlin-reflect:1.9.22") // Sapphire
+
     // Pufferfish start
     implementation("org.yaml:snakeyaml:1.32")
     implementation ("me.carleslc.Simple-YAML:Simple-Yaml:1.8.4") {
