From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Mon, 2 May 2022 01:27:17 +0500
Subject: [PATCH] Allow force spawns

Ported from EmpireCraft

diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index aaa7ad2a14389dc0dbc0d0fa3fb5ea16ec4172f6..a436e9c543f0f1724868c82af8ef4bd1dba96ff3 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -1368,7 +1368,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
         // SPIGOT-6415: Don't call spawn event for entities which travel trough worlds,
         // since it is only an implementation detail, that a new entity is created when
         // they are traveling between worlds.
-        this.addDuringTeleport(entity, null);
+        this.addDuringTeleport(entity, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.FORCE); // Sapphire
     }
 
     public void addDuringTeleport(Entity entity, CreatureSpawnEvent.SpawnReason reason) {
diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index 1d8ec0f85ec42f2dcd9405df83b526ae1c59de6f..23da36b6377f34a56ed3900ab9b55230892606d3 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -702,7 +702,7 @@ public class CraftEventFactory {
             event = CraftEventFactory.callEntitySpawnEvent(entity);
         }
 
-        if (event != null && (event.isCancelled() || entity.isRemoved())) {
+        if (spawnReason != SpawnReason.FORCE && event != null && (event.isCancelled() || entity.isRemoved())) { // Sapphire
             Entity vehicle = entity.getVehicle();
             if (vehicle != null) {
                 vehicle.discard();
