From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Sun, 31 Jul 2022 17:19:55 +0500
Subject: [PATCH] Disable timings by default


diff --git a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
index a7f4b791894c5d8b7af4b0c067cf167b29272a27..2f7d17d5f3860cee962886212d28eaa46f9a1d20 100644
--- a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
+++ b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
@@ -52,7 +52,7 @@ public class GlobalConfiguration extends ConfigurationPart {
     public Timings timings;
 
     public class Timings extends ConfigurationPart.Post {
-        public boolean enabled = true;
+        public boolean enabled = false; // Sapphire - disable timings by default
         public boolean reallyEnabled = false;
         public boolean verbose = true;
         public String url = "https://timings.aikar.co/";
