From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DenaryDev <denaryplanet@gmail.com>
Date: Mon, 2 May 2022 01:17:40 +0500
Subject: [PATCH] Add Player.getFishingHook API

Ported from EmpireCraft

diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index e50f443783ba728b73f25f029ce881255e345182..56b4dd386e1c6869897ac8e5c7c3d85a26a92d5b 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2783,11 +2783,19 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
     }
     // Paper end
 
-    // Sapphire start - Chromium client support
+    // Sapphire start
     @Override
     public boolean usesChromiumClient() {
         return getHandle().chromiumClient;
     }
+
+    @Override
+    public org.bukkit.entity.FishHook getFishingHook() {
+        if (getHandle().fishing != null) {
+            return (org.bukkit.entity.FishHook) getHandle().fishing.getBukkitEntity();
+        }
+        return null;
+    }
     // Sapphire end
 
     public Player.Spigot spigot()
